### ISSUE TITLE
memory.o: symbol changed sections: .LC39

### ISSUE THREAD

**Issue by disaster123** [dynup/kpatch#714]
Hello,

i'm trying to apply this simply looking patch on top of my 4.4 kernel via kpatch but build fails with:

...
Reading special section data
Building original kernel
Building patched kernel
Extracting new and modified ELF sections
ERROR: memory.o: symbol changed sections: .LC39
/usr/lib/kpatch/create-diff-object: unreconcilable difference
mmap.o: changed function: __split_vma.isra.38
mmap.o: changed function: copy_vma
mmap.o: changed function: unmapped_area_topdown
mmap.o: changed function: SyS_remap_file_pages
mmap.o: changed function: unmapped_area
ERROR: 1 error(s) encountered. Check /root/.kpatch/build.log for more details.

Patch:
```
 mm/memory.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/mm/memory.c b/mm/memory.c
index 380b731..581cc73 100644
--- a/mm/memory.c
+++ b/mm/memory.c
@@ -2684,9 +2684,10 @@ static int do_anonymous_page(struct mm_struct *mm, struct vm_area_struct *vma,
 		return VM_FAULT_SIGBUS;
 
 	/* Check if we need to add a guard page to the stack */
-	if ((vma->vm_flags & (VM_GROWSDOWN|VM_GROWSUP)) &&
-			expand_stack(vma, address) < 0)
-		return VM_FAULT_SIGSEGV;
+	if (stack_guard_area(vma, address)) {
+		if (expand_stack(vma, address) < 0)
+			return VM_FAULT_SIGSEGV;
+	}
 
 	/* Use the zero-page for reads */
 	if (!(flags & FAULT_FLAG_WRITE) && !mm_forbids_zeropage(mm)) {
```

Greets,
Stefan

**Comment by rudis** [comment]
I came across a similar issue and the referenced patch fixes it for me (I think). Could you test my patch to see if it helps. I tried to find the kernel you used but couldn't find this hunk.

**Comment by disaster123** [comment]
It's a custom kernel based on the opensuse 42.3 kernel ;-)

**Comment by jpoimboe** [comment]
@disaster123 I checked the opensuse 42.3 kernel (kernel-default-4.4.79-19.1) but this patch doesn't apply to it either.

**Comment by disaster123** [comment]
@jpoimboe yes not to the current one the report was from June 21. So you have to use the git version from that date it might also be a 42.2 kernel. Sorry was 2 month ago ;-)

**Comment by jpoimboe** [comment]
@disaster123 Sorry we didn't get to this one earlier.  I wasn't able to find the kernel and recreate your issue, but I'm assuming it's fixed by #730.  If you can figure out how to recreate, please test with that patch.  Thanks.
